cms/djangoapps/contentstore/views/course.py

```diff
@@ -283,6 +293,32 @@ def course_rerun_handler(request, course_key_string):
             })
 
 
+@login_required
+@ensure_csrf_cookie
+@require_GET
+def course_index_handler(request, course_key_string):
+    """
+    The restful handler for course indexing.
+    GET
+        html: return status of indexing task
+    """
+    # Only global staff (PMs) are able to rerun courses during the soft launch
+    if not GlobalStaff().has_user(request.user):
+        raise PermissionDenied()
+    course_key = CourseKey.from_string(course_key_string)
+    with modulestore().bulk_operations(course_key):
+        error = reindex_course_and_check_access(course_key, request.user)
```

Comment:
If there's an error with the `reindex_course_and_check_access` function, I would expect it to raise an exception. Returning an error value is an anti-pattern in Python -- it means that the caller has to know to check for errors. Raising an exception means that the caller doesn't have to do anything unless it wants to explicitly handle the error. Why does `reindex_course_and_check_access` return an error value, rather than raising an exception on error?